src/http/Stream.cc

```diff
@@ -556,8 +554,7 @@ void
 Http::Stream::deferRecipientForLater(clientStreamNode *node, HttpReply *rep, StoreIOBuffer receivedData)
 {
     debugs(33, 2, "Deferring request " << http->uri);
-    assert(flags.deferred == 0);
-    flags.deferred = 1;
+    assert(!deferredparams.rep);
```

Comment:
For the transition from the flag to a pointer to preserve original semantics, we have to assert that the pointer is not nil. I have not checked whether all callers support this transition, but if they do, then something along these lines may work:

```suggestion
    Assure(!deferredparams.rep);
    Assure(rep);
```